src/full/Agda/Interaction/Options/Warnings.hs

```diff
@@ -421,7 +421,7 @@ warningNameDescription = \case
   UnsolvedMetaVariables_           -> "Unsolved meta variables."
   UserWarning_                     -> "User-defined warning added using one of the 'WARNING_ON_*' pragmas."
   WithoutKFlagPrimEraseEquality_   -> "`primEraseEquality' usages with the without-K flags."
-  WrongInstanceDeclaration_        -> "Terms marked as eligible for instance search should end with a name."
+  WrongInstanceDeclaration_        -> "Instances should not be functions."
```

Comment:
Mmh, not sure that "not functions" is the right qualification here...
How about: 
> Instances that do not adhere to the required format.